packages/yarnpkg-core/sources/LegacyMigrationResolver.ts

```diff
@@ -31,6 +31,17 @@ export const IMPORTED_PATTERNS: Array<[RegExp, (version: string, ...args: Array<
   [/^[^/]+\.tgz#[0-9a-f]+$/, version => `npm:${version}`],
 ];
 
+function getPotentialDescriptor(descriptor: Descriptor) {
+  try {
+    const parsedRange = structUtils.parseRange(descriptor.range);
+    const potentialDescriptor = structUtils.tryParseDescriptor(parsedRange.selector, true);
+    if (potentialDescriptor) {
+      return potentialDescriptor;
+    }
```

Comment:
```suggestion
    return potentialDescriptor || undefined;
```

This can be simplified (not a big problem though)

Reply:
I just realized explicit return for undefined is not required on line 42.
I'll remove that instead for simplicity, and retain potentialDescriptor only if it's defined.